target/scripts/build/packages.sh

```diff
@@ -132,16 +132,31 @@ function _install_dovecot
 function _install_rspamd
 {
   _log 'trace' 'Adding Rspamd package signatures'
-  curl -sSfL https://rspamd.com/apt-stable/gpg.key | gpg --dearmor >/etc/apt/trusted.gpg.d/rspamd.gpg
-
-  echo "deb [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/rspamd.gpg] http://rspamd.com/apt-stable/ bullseye main" \
-    >/etc/apt/sources.list.d/rspamd.list
-  echo "deb-src [arch=amd64 signed-by=/etc/apt/trusted.gpg.d/rspamd.gpg] http://rspamd.com/apt-stable/ bullseye main" \
-    >>/etc/apt/sources.list.d/rspamd.list
+  local DEB_FILE='/etc/apt/sources.list.d/rspamd.list'
+  local RSPAMD_PACKAGE_NAME
+
+  # We try getting the most recent version of Rspamd for aarch64 (from an official source, which
+  # is the backports repository). The version for aarch64 is 3.2; the most recent version for amd64
+  # that we get with the official PPA is 3.4.
+  #
+  # Not removing it later is fine as you have to explicitly opt into installing a backports package
+  # which is not something you could be doing by accident.
+  if [[ $(uname --machine) == 'aarch64' ]]
```

Comment:
Not a big fan of this due to version difference, and the official rspamd advice is to not use the debian packages :sweat_smile: 

Hopefully the version disparity doesn't cause issues going forward if we take the approach you've chosen to support ARM64.

---

I had a browse of the [rspamd download docs](https://rspamd.com/downloads.html), and noticed they have support for Fedora (_and OpenSUSE via OBS, but TW isn't a good idea, and Leap releases are slow - not that Debian is any better_). 

At one point I wanted to consider Alpine as a base image offering, especially for the regular releases and newer packages, but I'm more reluctant now with all the `musl` compatibility and performance issues I've seen with it :sweat_smile: 

Fedora might be interesting, they have a minimal base image (slightly bigger than Debian), but hard to say if it's worthwhile or a smooth transition to change to. Those rspamd docs made it seem like it was supported but [it's been broken for a while](https://github.com/rspamd/rspamd.com/issues/604).

Maybe [OpenSUSE Leap image base](https://hub.docker.com/r/opensuse/leap/tags) would be worth looking into. It's fairly small at a glance, but I understand there is more familiarity with Debian :sweat_smile:  (_Leap is not likely to stay around either AFAIK as they're working towards a new container focused distro/spin_)

---

One other option is building rspamd ourselves with a multi-stage build, but that has it's own drawbacks (_maintenance + build time + CI cache?_).

Reply:
I agree, it's not very nice, but as of now, ARM64 users would have to use v2.7. With this change, they can use v3.2 which is only 2 minor versions behind the official PPA. I will rework this as soon as a better emerges :)

---

About the base image change: we should put this into our backlog for now I think :)

Reply:
> About the base image change: we should put this into our backlog for now I think :)

I am in no rush, that's not a fun change to tackle and really needs to be worthwhile to bother with :sweat_smile: 

My main motivations were attempting to reduce image weight and switch to something with more frequent release cadence for us to not rely on all these external builds for newer versions, as reliance on that seems to be growing.

---

Debian Bookworm release should bump up the rspamd package release at least? :)

I [raised a request to get official ARM64 builds supported](https://github.com/rspamd/rspamd/issues/4390), but given the history of maintainer stances on that, I'm not sure if it'll be approved on their end.

I've also further investigated the Fedora situation further and no fedora release going back 5 years has support to install `rspamd` based on their instructions :confused: Their docs are terribly out of date for that support it seems.

Reply:
This is going to be a tough one indeed. Ubuntu probably wont help here either as it's basically Debian.. We should discuss about this first on a maintainers discussion board or with discussions for this repository so users can post too. 

I thought Fedora was actually quite nice, while I don't have a great experience with OpenSUSE, but I'm open to listening to all proposals.

Reply:
No rush, we'll likely update to Bookworm before we change distro :)